TestsOutsidePackage/test_score.py

```diff
@@ -54,3 +93,272 @@ def test_run_scoring(test_output_dirs: OutputFolderForTests, is_ensemble: bool)
+def test_unpack_flat_zip(zip_filename: Path, test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test that a zip file containing just files: 1.txt, 2.dcm, 3.dcm, 4.dcm and 5.txt
+    can be extracted into a folder containing only the .dcm files.
+
+    :param zip_filename: Path to test zip file.
+    :param test_output_dirs: Test output directories.
+    """
+    _common_test_unpack_dicom_zip(zip_filename, TEST_FLAT_ZIP_FILENAMES, test_output_dirs)
+
+
+@pytest.mark.parametrize("zip_filename", [TEST_TWO_ZIP_FILE, TEST_TWO_NESTED_ZIP_FILE, TEST_TWO_NESTED_TWICE_ZIP_FILE])
+def test_unpack_two_zip(zip_filename: Path, test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test that a zip file containing files: 1.txt, 2.dcm, 3.dcm, 4.dcm and 5.txt in one folder
+    and 5.txt, 6.dcm, 7.dcm, 8.dcm, 9.txt in another folder
+    can be extracted into a folder containing only the .dcm files.
+
+    :param zip_filename: Path to test zip file.
+    :param test_output_dirs: Test output directories.
+    """
+    _common_test_unpack_dicom_zip(zip_filename, TEST_TWO_ZIP_FILENAMES, test_output_dirs)
+
+
+def _common_test_unpack_dicom_zip(zip_filename: Path, expected_filenames: List[str],
+                                  test_output_dirs: OutputFolderForTests) -> None:
+    """
+    Test the zip file contains expected .dcm files not in a folder.
+
+    :param zip_filename: Path to test zip file.
+    :param expected_filenames: List of expected filenames.
+    :param test_output_dirs: Test output directories.
+    """
+    model_folder = test_output_dirs.root_dir / "unpack"
+    extraction_folder = model_folder / "temp_zipped_dicom_extraction"
+    extract_zipped_dicom_series(zip_filename, extraction_folder)
+    _common_test_unpack_zip(extraction_folder, expected_filenames)
+
+
+def _common_test_unpack_zip(extraction_folder: Path, expected_filenames: List[str]) -> None:
+    """
+    Test the extraction folder contains expected files, not in a folder.
+
+    :param extraction_folder: Path to extracted zip file.
+    :param expected_filenames: List of expected filenames.
+    """
+    # Get all files/folders in this series
+    extracted_files = list(extraction_folder.glob('**/*'))
+    # Check they are all files (no folders) and in the extraction_folder (not a subdirectory)
+    for extracted_file in extracted_files:
+        assert extracted_file.is_file()
+        relative_path = extracted_file.relative_to(extraction_folder)
+        assert str(relative_path.parent) == '.'
+    extracted_file_names = sorted([extracted_file.name for extracted_file in extracted_files])
+    # Check names are as expected
+    assert extracted_file_names == expected_filenames
+
+
+def assert_zip_file_contents(zip_filename: Path, expected_filenames: List[str],
+                             model_folder: Path) -> None:
+    """
+    Compare the contents of a zip file, with a list of expected filenames.
```

Comment:
can you say more? expected files - should there be set equality to the files in zip, or is subset OK?